cmd/kops/apply.go

```diff
@@ -0,0 +1,134 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+	"k8s.io/kops/cmd/kops/util"
+	"k8s.io/kops/util/pkg/vfs"
+
+	"bytes"
+
+	kopsapi "k8s.io/kops/pkg/apis/kops"
+	"k8s.io/kubernetes/pkg/kubectl/cmd/templates"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/kubectl/resource"
+	"k8s.io/kubernetes/pkg/util/i18n"
+)
+
+var (
+	apply_long = templates.LongDesc(i18n.T(`
+		Apply a resource specification by filename or stdin.`))
+
+	apply_example = templates.Examples(i18n.T(`
+		# Apply a cluster specification using a file
+		kops apply -f my-cluster.yaml
+		`))
+
+	apply_short = i18n.T(`Apply cluster resources.`)
+)
+
+type ApplyOptions struct {
+	resource.FilenameOptions
+}
+
+func NewCmdApply(f *util.Factory, out io.Writer) *cobra.Command {
+	options := &ApplyOptions{}
+
+	cmd := &cobra.Command{
+		Use:     "apply -f FILENAME",
+		Short:   apply_short,
+		Long:    apply_long,
+		Example: apply_example,
```

Comment:
You can add an alias here for replace and then remove the replace code.  There is no need to have duplicate code.